packages/babel-generator/test/fixtures/parentheses/class-extends/expected.js

```diff
@@ -0,0 +1,35 @@
+class A extends (() => {}) {}
+
+class A extends (B = C) {}
+
+class A extends (B || C) {}
+
+class A extends (B + C) {}
+
+class A extends B() {}
+
+class A extends class {} {}
+
+class A extends (B ? C : D) {}
+
+class A extends (new B()) {}
+
+class A extends (B, C) {}
+
+class A extends ({}) {}
```

Comment:
Are the parentheses actually needed here?

Reply:
updated!